code/modules/mob/living/basic/lavaland/watcher/watcher_gaze.dm

```diff
@@ -0,0 +1,115 @@
+/**
+ * Do something nasty to everyone nearby if they're looking at us.
+ */
+/datum/action/cooldown/watcher_gaze
+	name = "Disorienting Gaze"
+	desc = "After a delay, flash everyone looking at you."
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "gaze"
+	background_icon_state = "bg_demon"
+	overlay_icon_state = "bg_demon_border"
+	cooldown_time = 30 SECONDS
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_INCAPACITATED
+	/// At what range do we check for vision?
+	var/effect_radius = 7
+	/// How long does it take to play our various animation stages
+	var/animation_time = 0.8 SECONDS
+	/// How long after pressing the button do we give people to turn around?
+	var/wait_delay = 1.6 SECONDS
+	/// What are we currently displaying?
+	var/image/current_overlay
+	/// Timer until we go to the next stage
+	var/stage_timer
+
+/datum/action/cooldown/watcher_gaze/Activate(mob/living/target)
+	. = ..()
+	show_indicator_overlay("eye_open")
+	stage_timer = addtimer(CALLBACK(src, PROC_REF(show_indicator_overlay), "eye_pulse"), animation_time, TIMER_STOPPABLE)
+	if (!do_after(owner, delay = wait_delay, target = owner))
+		deltimer(stage_timer)
+		clear_current_overlay()
+		return
+	trigger_effect()
```

Comment:
- I suggest not calling parent for `Activate` unless you intend on making use of ability sequences
- Be sure to call `StartCooldown()` and `return TRUE`
- Since this does a doafter, also start a dummy cooldown at the start so it can't be spammed (`StartCooldown(360 SECONDS, 360 SECONDS)`)